### DISCUSSION TITLE
Correct procedure to request a flag for a formula

### DISCUSSION THREAD

**Discussion by Etheryte** [Homebrew/discussions#25] [Everyday usage]
The formula for [Terraform](https://formulae.brew.sh/formula/terraform#default) includes [a hardcoded build-time env variable](https://github.com/Homebrew/homebrew-core/blob/master/Formula/terraform.rb#L30) that disables using the [cgo resolver](https://golang.org/pkg/net/#hdr-Name_Resolution):  

```rb
  def install
    ENV["CGO_ENABLED"] = "0"
    # ...
  end
```

The variable was introduced in [this commit](https://github.com/Homebrew/homebrew-core/commit/f0cd53a0d1f143b3faecdfc074c8d7cb4a88a6cf) which sadly doesn't document why this change was made.  

Using the cgo resolver can be a necessity when using Terraform with [Avahi](https://www.avahi.org/) for example.  

Currently I've manually edited the formula on my machine, but it would be nice to either revert this addition altogether or to add a flag so users can enable the resolver when installing the formula.

What's the correct procedure to file such a request?

**Comment by gromgit** [comment]
If you're convinced that this is the Right Thing to do for all Homebrew Terraform users, submit a PR to change it: https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request

**Comment by SMillerDev** [comment]
> Currently I've manually edited the formula on my machine, but it would be nice to either revert this addition altogether or to add a flag so users can enable the resolver when installing the formula.

Making a pull request is the right way to request changes in homebrew. There's not going to be an option to have it both ways though, Homebrew core prohibits the use of install flags in formula.

**Comment by Etheryte** [comment]
Thanks for your input, I'm hesitant to change the value outright because it's not clear why it was added in the first place, [Chesterton's fence](https://en.wikipedia.org/wiki/Wikipedia:Chesterton%27s_fence) and all, that's why I considered a flag to be solid middle ground. Since that isn't an option apparently, I'll try to figure out if there's any limitations to having it enabled. Currently it seems building etc works fine, but there may be more to it.
